src/rocksdb_options.rs

```diff
@@ -85,21 +86,9 @@ impl BlockBasedOptions {
         }
     }
 
-    /// the recommanded shard_bits is 6, also you can set a larger value as long as it is
-    /// smaller than 20, also you can set shard_bits to -1, RocksDB will choose a value for you
-    /// the recommanded capacity_limit is 0(false) if your memory is sufficient
-    /// the recommanded pri_ratio should be 0.05 or 0.1
-    pub fn set_lru_cache(
-        &mut self,
-        size: size_t,
-        shard_bits: c_int,
-        capacity_limit: c_uchar,
-        pri_ratio: c_double,
-    ) {
-        let cache = crocksdb_ffi::new_cache(size, shard_bits, capacity_limit, pri_ratio);
+    pub fn set_block_cache(&mut self, cache: Cache) {
         unsafe {
-            crocksdb_ffi::crocksdb_block_based_options_set_block_cache(self.inner, cache);
-            crocksdb_ffi::crocksdb_cache_destroy(cache);
```

Comment:
why before destroy here?

Reply:
We now pass in a `Cache` struct, which implements `Drop` trait, and will call `crocksdb_cache_destroy` implicitly.

Reply:
Before `cache` was a pointer to a native C struct.